torch/csrc/utils/object_ptr.h

```diff
@@ -8,9 +8,7 @@ class TORCH_PYTHON_API THPPointer {
  public:
   THPPointer() : ptr(nullptr){};
   explicit THPPointer(T* ptr) noexcept : ptr(ptr){};
-  THPPointer(THPPointer&& p) noexcept {
-    free();
-    ptr = p.ptr;
+  THPPointer(THPPointer&& p) noexcept : ptr(p.ptr) {
     p.ptr = nullptr;
   };
```

Comment:
```suggestion
  THPPointer(THPPointer&& p) noexcept : ptr(std::exchange(p.ptr, nullptr)) {}
```
is a bit more cleaner and does what you want I think.